src/app/_components/Gameboard/_subcomponents/Overlays/ChatDrawer/ChatDrawer.tsx

```diff
@@ -53,45 +87,83 @@ const ChatDrawer: React.FC<IChatDrawerProps> = ({ sidebarOpen, toggleSidebar })
-            color: '#FFF',
-            fontSize: '20px',
-            border: '1px solid transparent',
-            borderRadius: '10px',
-            pt: '10px',
-            pb: '10px',
-            justifyContent: 'space-between',
-            paddingLeft: '12px',
-            paddingRight: '35px',
-            position: 'relative',
-            '& .MuiButton-startIcon': {
-                marginRight: 0,
-                marginLeft: 0,
-                transform: 'skewX(5deg)',
-                display: { xs: 'none', sm: 'none', md: 'block' },
-                '& svg': {
-                    width: '23px',
-                    height: '23px',
-                },
-            },
+            background: 'linear-gradient(rgb(29, 29, 29), #0a3d1e) padding-box, linear-gradient(to top, #1cb34a, #0a3d1e) border-box',
             '&:hover': {
                 background: 'linear-gradient(rgb(29, 29, 29),rgb(20, 81, 40)) padding-box, linear-gradient(to top, #2ad44c, #0a3d1e) border-box',
                 boxShadow: '0 0 8px rgba(0, 170, 70, 0.7)',
                 border: '1px solid rgba(0, 200, 90, 0.7)',
             },
+        },
+        quickUndoButtonDisabled: {
+            ...quickUndoButtonBase,
+            width: 'min(55%, 200px)',
+            '&:disabled': {
+                backgroundColor: '#404040',
+                color: '#FFF'
+            },
+        },
+        quickUndoButtonBlocked: {
+            ...quickUndoButtonBase,
+            width: 'min(65%, 240px)',
             '&:disabled': {
                 backgroundColor: '#404040',
                 color: '#FFF'
             },
-            transform: 'skewX(-5deg)',
+        },
+        quickUndoButtonRequest: {
+            ...quickUndoButtonBase,
+            width: 'min(65%, 240px)',
+            background: 'linear-gradient(#1E2D32, #1E2D32) padding-box, linear-gradient(#404040, #008FC4) border-box',
+            '&:hover': {
+                background: 'linear-gradient(#2C4046, #2C4046) padding-box, linear-gradient(#404040, #008FC4) border-box',
+            },
         }
     }
-    
+
+    let undoButtonStyle;
+    switch (quickUndoState) {
+        case QuickUndoAvailableState.FreeUndoAvailable:
+            undoButtonStyle = styles.quickUndoButtonEnabled;
+            break;
+        case QuickUndoAvailableState.UndoRequestsBlocked:
+            undoButtonStyle = styles.quickUndoButtonBlocked;
+            break;
+        case QuickUndoAvailableState.RequestUndoAvailable:
+            undoButtonStyle = styles.quickUndoButtonRequest;
+            break;
+        default:
+            undoButtonStyle = styles.quickUndoButtonDisabled;
+            break;
+    }
+
+    let buttonText;
+    switch (quickUndoState) {
+        case QuickUndoAvailableState.RequestUndoAvailable:
+            buttonText = 'Request';
+            break;
+        case QuickUndoAvailableState.UndoRequestsBlocked:
+            buttonText = 'Blocked';
+            break;
+        default:
```

Comment:
Are we good with showing a normal undo button when there is no snapshot available?  Is it just a no-op?